### PR TITLE
feat: profile info upload, move leaderboard logic

### PR DESCRIPTION
**Pull request by jonowu**
## Purpose

Adds uploading of account info including profile picture.

## Changes

- Add support for profile pictures, and default to the user's initials if they have no picture.
- Add edit profile screen
- Moved leaderboard logic to its own screen
- Added typography components @BenjaminYong 

### Screenshots

Screen Shot 2020-10-06 at 8 55 44 pm
Screen Shot 2020-10-06 at 8 56 54 pm
Screen Shot 2020-10-06 at 8 57 23 pm

## Checklist

- [ ] Tested on Android
- [x] Tested on iOS

### CODE REVIEW

**Comment by BenjaminYong** [comment]
Uploading DP works great, however, after saving changes to a newly uploaded DP, the photo reverts to the previous DP within the customise screen. But once you go back to main profile screen, the changes are reflected

**Comment by BenjaminYong** [comment]
Just some general usability feedback, perhaps a hyperlink that says 'Change DP' or 'Change' in addition to the DP button in customise screen. 

Something like this:
image

**Comment by BenjaminYong** [comment]
Made a new user, tried to visit leaderboard, but an error breaks the app (undefined data.profilePicture)

**Comment by BenjaminYong** [comment]
> Made a new user, tried to visit leaderboard, but an error breaks the app (undefined data.profilePicture)
> 
> 

Still breaks when visiting leaderboard after uploading a DP, not sure what's going on

**Comment by jonowu** [comment]
Cheers for the testing and feedback @BenjaminYong, I've made changes to address all the points you made. Happy to make more changes if you think there could be improvements

**Comment by BenjaminYong** [comment]
Sweet all fixes work great! I like the edit pencil icon for the change DP 👍🏼

**Comment by BenjaminYong** [comment]
Found a new bug, sorry @jonowu haha 
Steps to recreate:
- Populate the three profile text fields then save.
- Delete one of these fields, i.e. Fun facts, then save
- Immediate error: Text strings must be rendered within a <Text> component
- Cannot rectify the error on the user's end, endless error loop as the data was saved.
- Maybe a condition to only render individual profile text section if not empty?

Some small UI feedback for the fun facts section in profile:
- Perhaps change 'Information' to 'Bio' like Instagram?
- The subtitle colours, especially the blue could be a UI issue due to the dark background - hard to see on low brightness.
    - Though, we should probably touch base with Tais to see what she thinks
- Could increase the DP size in profile to fill in blank space on the top, maybe an adjustable prop/param for that component?

**Comment by jonowu** [comment]
Cheers again @BenjaminYong for the thorough review, I have now:
- used lodash's `isEmpty()` utility to check for empty strings before rending the profile info
- made the profile picture bigger
- used a contrast checker to make the colours WCAG Level AAA compliant (https://webaim.org/resources/contrastchecker/)

Screen Shot 2020-10-14 at 8 00 04 pm

File: screens/edit-profile/index.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+              lastName={userContext.lastName}
+            />
+            <MaterialCommunityIcons
+              name="pencil-circle"
+              size={30}
+              style={{ position: 'absolute', top: 55, left: 45 }}
+            />
+          </TouchableOpacity>
+        </HeaderView>
+        <Input label="Information" value={information} onChangeText={(value) => setInformation(value)} />
```
Comment (BenjaminYong) [inline screens/edit-profile/index.js:146]: To update to bio in future PR

**Review by BenjaminYong — APPROVED** [review]
Looks good man, like the new accessible colors against the backdrop ;)

File: screens/edit-profile/index.js (hunk 1 above)
Comment (jonowu) [reply screens/edit-profile/index.js:146]: Ended up doing this anyway, just didn't rename any variables 😎

**Pull request merged**